atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/cache/ValueStoreImplTest.java

```diff
@@ -82,48 +89,92 @@ public void lockEventInvalidatesValue() {
+        assertExpectedValue(EXACT_ROW_TABLE, ROW_LEVEL_CELL, CacheEntry.locked());
+
+        verify(metrics).decreaseCacheSize(expectedSize);
+    }
+
     @Test
     public void unlockEventsClearLockedEntries() {
-        valueStore.applyEvent(WATCH_EVENT);
+        valueStore.applyEvent(WATCH_EVENTS);
         valueStore.applyEvent(LOCK_EVENT);
+        valueStore.applyEvent(ROW_LOCK_EVENT);
 
         assertExpectedValue(CELL_1, CacheEntry.locked());
+        assertExpectedValue(EXACT_ROW_TABLE, ROW_LEVEL_CELL, CacheEntry.locked());
 
         valueStore.applyEvent(UNLOCK_EVENT);
+        valueStore.applyEvent(ROW_UNLOCK_EVENT);
         assertThat(valueStore.getSnapshot().getValue(TABLE_CELL)).isEmpty();
+        assertThat(valueStore.getSnapshot().getValue(ROW_LEVEL_CELL_REFERENCE)).isEmpty();
+    }
+
+    @Test
+    public void resetClearsAllEntries() {
+        valueStore.applyEvent(WATCH_EVENTS);
+        valueStore.applyEvent(LOCK_EVENT);
+        valueStore.applyEvent(ROW_LOCK_EVENT);
+
+        assertExpectedValue(CELL_1, CacheEntry.locked());
+        assertExpectedValue(EXACT_ROW_TABLE, ROW_LEVEL_CELL, CacheEntry.locked());
+
+        valueStore.reset();
+        assertThat(valueStore.getSnapshot().getValue(TABLE_CELL)).isEmpty();
+        assertThat(valueStore.getSnapshot().getValue(ROW_LEVEL_CELL_REFERENCE)).isEmpty();
     }
 
     @Test
     public void putValueThrowsIfCurrentValueDiffers() {
-        valueStore.applyEvent(WATCH_EVENT);
+        valueStore.applyEvent(WATCH_EVENTS);
         valueStore.putValue(TABLE_CELL, VALUE_1);
+        valueStore.putValue(ROW_LEVEL_CELL_REFERENCE, VALUE_1);
 
         assertThatCode(() -> valueStore.putValue(TABLE_CELL, VALUE_1)).doesNotThrowAnyException();
+        assertThatCode(() -> valueStore.putValue(ROW_LEVEL_CELL_REFERENCE, VALUE_1))
+                .doesNotThrowAnyException();
         assertPutThrows(VALUE_2);
+        assertPutThrows(ROW_LEVEL_CELL_REFERENCE, VALUE_2);
 
         valueStore.applyEvent(LOCK_EVENT);
         assertPutThrows(VALUE_1);
+
+        valueStore.applyEvent(ROW_LOCK_EVENT);
+        assertPutThrows(ROW_LEVEL_CELL_REFERENCE, VALUE_1);
     }
 
     @Test
     public void watchEventUpdatesWatchableTables() {
-        assertThat(valueStore.getSnapshot().isWatched(TABLE)).isFalse();
-        valueStore.applyEvent(WATCH_EVENT);
-        assertThat(valueStore.getSnapshot().isWatched(TABLE)).isTrue();
+        assertThat(valueStore.getSnapshot().isWatched(ENTIRELY_WATCHED_TABLE)).isFalse();
+        assertThat(valueStore.getSnapshot().isWatched(ROW_LEVEL_CELL_REFERENCE)).isFalse();
+        valueStore.applyEvent(WATCH_EVENTS);
+        assertThat(valueStore.getSnapshot().isWatched(ENTIRELY_WATCHED_TABLE)).isTrue();
+        assertThat(valueStore.getSnapshot().isWatched(EXACT_ROW_TABLE)).isFalse();
+        assertThat(valueStore.getSnapshot().isWatched(ROW_LEVEL_CELL_REFERENCE)).isTrue();
     }
 
     @Test
     public void valuesEvictedOnceMaxSizeReached() {
```

Comment:
I didn't go through and add row-level references to this and the below tests, since they deal with overall value cache size, rather than things specific to entire-table or exact-row watches.